### ISSUE TITLE
Define rest api status code to return type mapping

### ISSUE THREAD

**Issue by sahejsingh** [anz-bank/sysl#244]
Currently while defining REST api using sysl, the return payload of the is interpreted as a string.

syslgen interprets the return payload as follows
- payload should be a comma separated list of names of types that the rest client expects to receive from the server
- the generated code tries to decode the received data in the order defined by the list

There is no support for HTTP status codes. Usually API's define that the server will return one type for 200 to <300 and Error type for others.

One possible solution is to use Oneof syntax
```
OneOf:
    statusCode == 200:
          return SuccessType
    statusCode == 400:
         return BadRequest
    statusCode == 401:
         return AuthError
```
OR
```
OneOf:
    statusCode >= 200 < 300:
          return SuccessType
    statusCode > 400:
         return Error
```

Whatever format we decide, syslgen will have to be able to translate that to target language.

**Comment by anzdaddy** [comment]
`OneOf` is allowable but shouldn't be required. My preference is to fold HTTP codes into the `return`, like so:
```
return 200 <: SuccessType
```
This allows any pseudocode structure, e.g.:
```
if Ready to process:
    loop over inputs:
        if failed:
            return 500 <: Error
    return 200 <: Results
```

**Comment by springwiz** [comment]
Addressed as part of PR #347.
